src/snapshotter.js

```diff
@@ -257,10 +259,21 @@ export default class SnapShotter {
       if (this._removeElements) await this.removeTheSelectors();
 
       if (this.wait) await this.snooze(this.wait);
+    } catch (err) {
+      this._onInfo();
+      logger.info(`Pre-snapshoot error: ${err}`);
+    }
+  }
+
+  async takeSnap() {
+    const filename = `${this._latest}/${this._label}-${
+      this._viewportLabel
+    }.png`;
 
-      const filename = `${this._latest}/${this._label}-${
-        this._viewportLabel
-      }.png`;
+    await this.driverSetup();
```

Comment:
if this fails, it would mean there is no connection to the grid and continuing to snap would not be possible.

Should return if this errors!

Reply:
^^^ ``` 
catch (err) {
      this._onError();
      logger.error(
        'snapshotter',
        `❌  Unable to connect to the grid at ${this._gridUrl}`
      );
      process.exitCode = 1;
      return;
    }```

Reply:
this is done in catch in driverSetup and i believe process.exitCode will fails if there is any error